Shared/Samples/Navigate route with rerouting/NavigateRouteWithReroutingView.Model.swift

```diff
@@ -0,0 +1,304 @@
+            
+            // Reset the location display.
+            await locationDisplay.dataSource.stop()
+            locationDisplay.autoPanMode = .off
+        }
+        
+        /// Resets the navigation.
+        func reset() async throws {
+            await stop()
+            
+            // Reset the graphics.
+            statusMessage = .initialInstructions
+            traversedRouteGraphic.geometry = nil
+            traversedRouteBuilder.replaceGeometry(with: nil)
+            
+            // Reset the route.
+            routeTracker.cancelRerouting()
+            simulatedDataSource.currentLocationIndex = 0
+            initializeProgressTracking()
+        }
+        
+        /// Updates the status message and route graphics using the progress from a given tracking status.
+        /// - Parameter status: The `TrackingStatus`.
+        func updateProgress(using status: TrackingStatus) async {
+            // Update the route graphics.
+            remainingRouteGraphic.geometry = status.routeProgress.remainingGeometry
+            
+            if let currentPosition = locationDisplay.location?.position {
+                traversedRouteBuilder.add(currentPosition)
+                traversedRouteGraphic.geometry = traversedRouteBuilder.toGeometry()
+            }
+            
+            // Update the status message.
+            switch status.destinationStatus {
+            case .approaching, .notReached:
+                // Format the route's remaining distance and time.
+                let distanceRemainingText = status.routeProgress.remainingDistance.distance.formatted()
+                
+                let dateInterval = DateInterval(start: .now, duration: status.routeProgress.remainingTime)
+                let dateRange = dateInterval.start..<dateInterval.end
+                let timeRemainingText = dateRange.formatted(
+                    .components(style: .abbreviated, fields: [.day, .hour, .minute, .second])
+                )
+                
+                statusMessage = """
+                Distance remaining: \(distanceRemainingText)
+                Time remaining: \(timeRemainingText)
+                """
+                
+                // Get the next direction from the route's direction maneuvers.
+                let nextManeuverIndex = status.currentManeuverIndex + 1
+                if route.directionManeuvers.indices.contains(nextManeuverIndex) {
+                    let nextDirection = route.directionManeuvers[nextManeuverIndex].text
+                    statusMessage.append("\nNext direction: \(nextDirection)")
+                }
+                
+            case .reached:
+                if status.remainingDestinationCount > 1 {
+                    statusMessage = "Intermediate stop reached, continue to next stop."
+                    try? await routeTracker.switchToNextDestination()
+                } else {
+                    await locationDisplay.dataSource.stop()
```

Comment:
When the destination is reached, it would be better to call reset. Otherwise I can still tap the pause/play button, and start a new navigation with the old traversed route graphic still being present.